frontend/src/lib/type.utils.ts

```diff
@@ -1,9 +1,28 @@
+import type { Get, IfNever } from 'type-fest';
+
 /**
- * creates a union of all possible nested keys of an object
+ * Removes properties with value `never` from an object type
  */
-export type NestedKeyOf<ObjectType extends object> =
+type OmitNever<T> = { [K in keyof T as IfNever<T[K], never, K>]: T[K] };
+
+/**
+ * Creates a union of all possible deep paths / nested keys (e.g. `obj.nestedObj.prop`) of an object
+ */
+export type DeepPaths<ObjectType extends object> =
   {
     [Key in keyof ObjectType & (string)]: ObjectType[Key] extends object
-    ? `${Key}` | `${Key}.${NestedKeyOf<ObjectType[Key]>}`
+    ? `${Key}` | `${Key}.${DeepPaths<ObjectType[Key]>}`
     : `${Key}`
   }[keyof ObjectType & (string)];
+
+/**
+ * Create a union of all possible deep paths of an object who's value fulfill `Condition`
+ */
+export type ConditionalDeepPaths<Base, Path extends string, Condition> = keyof OmitNever<{
```

Comment:
this is cool. Though I'm not sure I like the name. It is sorta a condition, but as a user that's not how I think about using it. I just want fields matching this type that I pass in. Maybe like `PathsExtending`? idk. Condition seems like a weird name as it's going to be an object type.

Reply:
I see now that I actually didn't make any usage of `ConditionalPick` from "type-fest" for this stuff. I went with "Conditional", because type-fest uses the term that way, but I agree that it's odd.
I'm gonna go with `DeepPathsToType`. "To" makes sense to me, because it's not the paths themselves that extend a type or match a condition.